src/libstd/ffi/c_str.rs

```diff
@@ -149,8 +209,13 @@ pub struct CStr {
 }
 
 /// An error returned from [`CString::new`] to indicate that a nul byte was found
-/// in the vector provided.
+/// in the vector provided.  While Rust strings may contain nul bytes in the middle,
+/// C strings can't, as that byte would effectively truncate the string.
```

Comment:
Could we change this up a bit? We try to have a summary sentence first, then the rest of it. This one has a *long* summary, and repeats itself since you added the information below. How about:

```
/// An error indicating that an interior nul byte was found.
///
/// While Rust strings may contain nul bytes in the middle, C strings can't, as that byte would effectively
/// truncate the string.
///
/// This `struct`....
```

with the correct wrapping, I just guessed here. What do you think?